compaction/compaction_manager.cc

```diff
@@ -1127,12 +1127,21 @@ future<> compaction_manager::drain() {
 }
 
 future<> compaction_manager::stop() {
-    do_stop();
-    if (auto cm = std::exchange(_task_manager_module, nullptr)) {
-        co_await cm->stop();
-    }
-    if (_state != state::none) {
-        co_return co_await std::move(*_stop_future);
+    try {
+        do_stop();
+        if (auto cm = std::exchange(_task_manager_module, nullptr)) {
+            auto f = cm->stop();
```

Comment:
```suggestion
            auto f = co_await cm->stop();
```
Before the change we await on `stop()` and we return whatever there is in `stop_future`.
After the change we await on `stop()` only if there was something in `stop_future` - otherwise we don't wait on `stop()`.

Reply:
> Before the change we await on stop() and we return whatever there is in stop_future.

The return type is `furure<>` so we don't really return anything (in _stop_future).

> After the change we await on stop() only if there was something in stop_future - otherwise we don't wait on stop().

why?
The change moves the future returned by cm->stop() into _stop_future and then it is co_await'ed as before.